kamelet-support/src/main/java/io/kaoto/backend/metadata/parser/step/camelRoute/CamelRouteFileProcessor.java

```diff
@@ -0,0 +1,315 @@
+
+        List<String> requiredProperties = propertiesParsed
+                .entrySet().stream().filter(
+                    property -> Boolean.parseBoolean(
+                            property.getValue().getOrDefault(
+                                    "required", "false"))
+        ).map(
+                Map.Entry::getKey
+        ).toList();
+
+
+        List<Parameter> parameters = propertiesParsed.entrySet().stream().map(
+                parameter -> {
+                    Map<String, String> parameterData = parameter.getValue();
+
+                    Map<String, Function<Map.Entry<
+                            String, Map<String, String>>,
+                            Parameter>> typeToClassConversion = Map.of(
+                            "string", this::getStringParameter,
+                            "object", this::getObjectParameter,
+                            "integer", this::getNumberParameter,
+                            "boolean", this::getBooleanParameter
+                    );
+
+                    Parameter parsedParameter = typeToClassConversion
+                            .getOrDefault(parameterData.get("type"),
+                                    this::getObjectParameter)
+                            .apply(parameter);
+
+                    if ("path".equals(parameterData.get("kind"))) {
+                        parsedParameter.setPath(true);
+                    }
+
+                    return parsedParameter;
+                }
+        ).toList();
+
+        return new JsonCamelObject(
+                id,
+                camelKind,
+                defaultIcon,
+                title,
+                description,
+                defaultGroup,
+                type,
+                parameters,
+                requiredProperties
+        );
+    }
+
+    private StringParameter getStringParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        StringParameter parsedParameter = new StringParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                parameterData.getOrDefault("defaultValue", null),
+                null
+        );
+
+        return parsedParameter;
+    }
+
+    private NumberParameter getNumberParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        //IntegerParameter doesn't fit some long types there are present
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        NumberParameter parsedParameter = new NumberParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                Long.parseLong(parameterData
+                        .getOrDefault("defaultValue", "0")
```

Comment:
Why? I'd rather have a default value if I know the component should have it if none is defined. It would be blank or 0 in a graphical user interface, waitiing for the user to fill it.

Reply:
Plus obviously to avoid the cases where no default value is given, and this is cleaner than having an if checking for it

Reply:
If none is defined, let it be null.

Reply:
it is important to keep the "no defined value". Here the code is making the assumption that no value will be considered as a 0 in the code. There is no way to ensure that, especially given that everybody can provide a Camel component.

Reply:
I'll put it at null then. You have a point.